src/fr/poseidonscans/src/eu/kanade/tachiyomi/extension/fr/poseidonscans/PoseidonScans.kt

```diff
@@ -1,14 +1,569 @@
+            ?: throw Exception("Could not extract Next.js data for chapter list.")
+
+        val mangaDetailsJson = pageData["manga"]?.jsonObject
+            ?: pageData["initialData"]?.jsonObject?.get("manga")?.jsonObject
+            ?: pageData.takeIf { it.containsKey("slug") && it.containsKey("title") }
+            ?: throw Exception("JSON 'manga' structure not found for chapters. JSON: ${pageData.toString().take(500)}")
+
+        val mangaDto = try {
+            Json { ignoreUnknownKeys = true }.decodeFromString<MangaDetailsData>(mangaDetailsJson.toString())
+        } catch (e: Exception) {
+            throw Exception("Error parsing chapters: ${e.message}. JSON: $mangaDetailsJson")
+        }
+
+        return mangaDto.chapters
+            ?.filter { it.isPremium != true }
+            ?.mapNotNull { ch ->
+                val chapterNumberString = ch.number.toString().removeSuffix(".0")
+                SChapter.create().apply {
+                    name = ch.title?.takeIf { it.isNotBlank() } ?: "Chapitre $chapterNumberString"
+                    url = "/serie/${mangaDto.slug}/chapter/$chapterNumberString"
+                    date_upload = parseIsoDate(ch.createdAt)
+                    chapter_number = ch.number
+                }
+            }
+            ?.sortedByDescending { it.chapter_number }
+            ?: emptyList()
+    }
+
+    private fun parseIsoDate(dateString: String?): Long {
+        return try {
+            if (dateString.isNullOrBlank()) return 0L
+            val cleanedDateString = if (dateString.startsWith("\"\$D")) {
+                dateString.removePrefix("\"\$D").removeSuffix("\"")
+            } else if (dateString.startsWith("\$D")) {
+                dateString.removePrefix("\$D")
+            } else if (dateString.startsWith("\"") && dateString.endsWith("\"") && dateString.length > 2) {
+                dateString.substring(1, dateString.length - 1)
+            } else {
+                dateString
+            }
+            isoDateFormatter.parse(cleanedDateString)?.time ?: 0L
+        } catch (e: Exception) {
+            0L
+        }
+    }
+
+    override fun pageListParse(document: Document): List<Page> {
+        val pageData = extractNextJsPageData(document)
+            ?: throw Exception("Could not extract Next.js data for page list.")
+
+        val chapterPageUrl = document.location()
+
+        val imagesListJson = pageData["images"]?.jsonArray
+            ?: pageData["chapter"]?.jsonObject?.get("images")?.jsonArray
+            ?: pageData["initialData"]?.jsonObject?.get("images")?.jsonArray
+            ?: pageData["initialData"]?.jsonObject?.get("chapter")?.jsonObject?.get("images")?.jsonArray
+            ?: throw Exception("JSON 'images' structure not found. Data: ${pageData.toString().take(500)}")
+
+        val imagesDataList = try {
+            Json { ignoreUnknownKeys = true }.decodeFromString<List<PageImageUrlData>>(imagesListJson.toString())
```

Comment:
don't initialize a new Json instance here and in other places. try using `.parseAs` from `keiyoushi.utils` instead